### PR TITLE
Losses

### PR DESCRIPTION
**Pull request by YannDubs**
- Modularized the losses. 
- added VAE loss / betaB / betaH 

PS: I didn't add Gaussian reconsstrction error for color images (i.e MSELoss) because I don't think it makes sense (although everyon uses it) because of what I said last time (every color channel i bounded by 255). So I renormalize and use binary cross entropy channel wise, which I think will work better. But we have to wait for the colored data to test. 

@linesd the framework for adding new losses is now done, you can have a look to know how to create the 2 new losses.

@alecokas for the experiments you can now decide on the loss function to use

### CODE REVIEW

File: disvae/losses.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    """
+    latent_dim = mean.size(1)
+    # batch mean of kl for each latent dimension
+    latent_kl = (-0.5 * (1 + logvar - mean.pow(2) - logvar.exp())).mean(dim=0)
+    total_kl = latent_kl.sum()
+
+    if storer is not None:
+        storer['kl_loss'].append(total_kl.item())
+        for i in range(latent_dim):
+            storer['kl_loss_' + str(i)].append(latent_kl[i].item())
```
Comment (alecokas) [inline disvae/losses.py]: As you noted in PR #5 there will be a conflict here but this should be no issue

**Comment by YannDubs** [comment]
mhh that's what I did actually (removed the .gif on github) and added `imgs/*` to gitignore

**Comment by linesd** [comment]
Thanks, looking at this now...
